.gitmodules

```diff
@@ -0,0 +1,6 @@
+[submodule "ceph_s3select"]
+	path = submodules/ceph_s3select
+	url = https://github.com/ceph/s3select
```

Comment:
add an exact commit/branch ref to pin the dependency.

Reply:
I'm went over https://git-scm.com/docs/gitmodules and didn't see relevant option.
Am I missing something?

Reply:
in gitmodules we can use the branch attribute.

if there is no upstream branch we want to track, and we need a specific sha we should use this method - https://stackoverflow.com/questions/8813249/git-submodules-specify-a-specific-sha